### ISSUE TITLE
Propose FindJulia.cmake to upstream cmake

### ISSUE THREAD

**Issue by termoshtt** [JuliaInterop/libcxxwrap-julia#28]
Hi team,

I try to use cmake for linking libjulia.so, and I need `find_package` for Julia. `FindJulia.cmake` in this repository is latest and well managed as far as I searched. Do you have a plan to propose this script to the [official cmake repository](https://gitlab.kitware.com/cmake/cmake)? It will be useful for other Julia users trying to use it from another language e.g. [Fortran](https://github.com/termoshtt/forjl).

Thanks.

**Comment by traversaro** [comment]
A possible alternative option (if there is someone willing to maintain it) is to install a `JuliaConfig.cmake` configuration file with Julia itself. Even if Julia is not build with CMake, it is possible nevertheless to generate a `JuliaConfig.cmake` file, see for example libxml2: 
* https://github.com/GNOME/libxml2/blob/619534ef2ed501e5206915064a5bc36153a897f7/Makefile.am#L1263
* https://github.com/GNOME/libxml2/blob/619534ef2ed501e5206915064a5bc36153a897f7/libxml2-config.cmake.in

**Comment by barche** [comment]
Yes, I think adding a `JuliaConfig.cmake` file would be the way to go. The best solution is to convert Julia to using CMake, but that's slightly more involved ;)

**Comment by SylvainCorlay** [comment]
Yes, it would be great if Julia moved to CMake.
